cpp/ql/lib/semmle/code/cpp/ir/implementation/aliased_ssa/Operand.qll

```diff
@@ -401,11 +407,19 @@ class ArgumentOperand extends RegisterOperand {
 }
 
 /**
- * An operand representing the implicit 'this' argument to a member function
+ * An operand representing the implicit `this` argument to a member function
  * call.
  */
 class ThisArgumentOperand extends ArgumentOperand {
   override ThisArgumentOperandTag tag;
+
+  // in most cases the def location makes more sense, but in some corner cases it
+  // does not have a location: in those cases we fall back to the use location
+  override Language::Location getLocation() {
+    if exists(Language::Location loc | loc = this.getAnyDef().getLocation())
+    then result = this.getAnyDef().getLocation()
+    else result = this.getUse().getLocation()
```

Comment:
This is a bit more idiomatic, I think. And it's a bit shorter :)
```suggestion
    result = this.getAnyDef().getLocation()
    or
    not exists(this.getAnyDef().getLocation()) and
    result = this.getUse().getLocation()
```

Reply:
actually I just found out this is not doing what I expected it to :/ I needed to check if the location is an instance of UnknownLocation (which still exists as a location).